### PR TITLE
Using ShardUpdateFields in reparent in a few places.

### PR DESCRIPTION
**Pull request by alainjobart**
Will be more resilient to underlying shard updates.

@michael-berlin @enisoc

### CODE REVIEW

File: go/vt/wrangler/tablet.go [hunk 1]
```diff
@@ -120,22 +121,21 @@ func (wr *Wrangler) DeleteTablet(ctx context.Context, tabletAlias *topodatapb.Ta
...
-		si, err := wr.ts.GetShard(ctx, ti.Keyspace, ti.Shard)
-		if err != nil {
+		// update the shard record's master
+		echoWarning := false
+		if _, err := wr.ts.UpdateShardFields(ctx, ti.Keyspace, ti.Shard, func(s *topodatapb.Shard) error {
+			if topoproto.TabletAliasEqual(s.MasterAlias, tabletAlias) {
+				s.MasterAlias = nil
+				echoWarning = false
+			} else {
+				echoWarning = true
```
Comment (enisoc) [inline go/vt/wrangler/tablet.go]: If the master alias is not the one we're looking for, I think we should return an error from the update() func. In that case, we aren't trying to change anything in the record, so there's no reason to let UpdateShardFields proceed, and there's no reason to retry if the version happened to go bad.

In that case, you can simplify this by logging the warning here.

**Comment by michael-berlin** [comment]
LGTM after Anthony's comment is addressed

**Comment by alainjobart** [comment]
PTAL I have changed the logic a bit, in a way that I can also use it in other places...

**Comment by enisoc** [comment]
LGTM thanks for the fix, it's really clear now.

**Pull request merged**